src/lib.rs

```diff
@@ -6,6 +6,26 @@
 //! * [`Mutex`] - a mutual exclusion lock.
 //! * [`RwLock`] - a reader-writer lock, allowing any number of readers or a single writer.
 //! * [`Semaphore`] - limits the number of concurrent operations.
+//!
+//! ## Relationship with `std::sync`
+//!
+//! In general, you should prefer to use [`std::sync`] types over types from this crate.
+//!
+//! There are two primary use cases for types from this crate:
+//!
+//! - You need to use a synchronization primitive in a `no_std` environment.
+//! - You need to hold a lock across an `.await` point.
+//!
+//! If you already use `libstd` and you aren't holding locks across await points, you should use
+//! [`std::sync`] instead of this crate. These types are optimized for operating system use cases,
```

Comment:
```suggestion
//! If you already use `libstd` and you aren't holding locks across await points, you should consider
//! [`std::sync`] instead of this crate. Those types are optimized for operating system use cases,
```